src/main/webapp/app/exercises/shared/exercise-scores/exercise-scores.component.html

```diff
@@ -238,205 +262,86 @@ <h4 *ngIf="this.rangeFilter" class="d-flex align-items-center">
-                        </a>
-                        <a
-                            *ngIf="
-                                exercise.exerciseGroup &&
-                                exercise.exerciseGroup!.exam?.numberOfCorrectionRoundsInExam !== undefined &&
-                                exercise.exerciseGroup!.exam!.numberOfCorrectionRoundsInExam! < 2
-                            "
-                            [routerLink]="[
-                                '/course-management',
-                                exercise.exerciseGroup!.exam!.course!.id,
-                                'exams',
-                                exercise.exerciseGroup!.exam!.id,
-                                'exercise-groups',
-                                exercise.exerciseGroup.id,
-                                'text-exercises',
-                                exercise.id,
-                                'participations',
-                                value.participation.id,
-                                'submissions',
-                                value.submission.id,
-                                'assessment'
-                            ]"
-                            class="btn btn-primary btn-sm me-1"
-                        >
-                            <fa-icon [icon]="faFolderOpen" [fixedWidth]="true"></fa-icon>
-                            {{ 'artemisApp.textSubmission.view' | artemisTranslate }}
-                        </a>
+                        <div class="button-header">
+                            <div class="mb-2" *ngIf="exercise.type === ExerciseType.PROGRAMMING">
+                                <a *ngIf="getRepositoryLink(value)" class="btn btn-primary btn-sm me-1" href="{{ getRepositoryLink(value) }}">
+                                    <fa-icon [icon]="faCodeBranch" class="me-1"></fa-icon>
+                                    Repository
+                                </a>
+                                <a *ngIf="buildPlanId(value)" target="_blank" rel="noreferrer" href="{{ value.buildPlanUrl }}" class="btn btn-primary btn-sm me-1">
+                                    <fa-icon class="me-1" [icon]="farFileCode"></fa-icon>
+                                    Build plan
+                                </a>
+                                <a
+                                    [routerLink]="['/courses', course.id, 'programming-exercises', exercise.id, 'code-editor', value.id]"
+                                    class="btn btn-info btn-sm me-1"
+                                    [jhiFeatureToggleLink]="FeatureToggle.ProgrammingExercises"
+                                >
+                                    <fa-icon class="me-1" [icon]="faFolderOpen" [fixedWidth]="true"></fa-icon>
+                                    Online editor
+                                </a>
+                            </div>
+                            <div>
+                                <a
```

Comment:
The submissions table had a green button ‘Assess Submissions’ in case no manual submission existed, and changed that to a blue one ‘View Assessment’ in case one did. It also showed ’Continue Assessment’/’Cancel Assessment’ for locked but unsubmitted ones.

Can you port this functionality over to the new table (maybe extract it into its own small component in case it wasn’t already)? That feature was quite useful for us when not using double-blind reviews. Ideally it would also show the reviewer.

Reply:
There is still the option to get to the assessment of each result as well as start one. I don't really see any benefit of distinguishing between the two cases, do you?

How often did you use the cancel/continue functionality?
- Continuing is easier from the assessment overview in my opinion or did you specifically search for a student where you wanted to continue the assessment?
- Canceling is readded :+1:

Reply:
> There is still the option to get to the assessment of each result as well as start one. I don't really see any benefit of distinguishing between the two cases, do you?

With the different colour it was immediately obvious if the student already has gotten a manual assessment, or is missing one. Before it was kind-of redundant since the result type and the assessor were shown in additional columns as well, but now. Now this information is not presented any more, unless you open the feedback view for the result to check if manual feedback exist.
The cases with an unfinished assessment described below would become immediately obvious when the different states were visually different.

> Continuing is easier from the assessment overview in my opinion or did you specifically search for a student where you wanted to continue the assessment?

Yes, we do not resolve plagiarism cases via Artemis but require students to come in person to the office to discuss those matters. In those cases an instructors locks those two submissions by starting the assessment, but not finishing it, so that it does not show up to other assessors and the assessments can continue for other submissions. Only after the 
discussion exactly those two assessments need to be continued and submitted.

We had at least once were where some tutor started an assessment but forgot to submit it. Then they were not reachable (e.g. holidays) and someone else had to look over and finish it. In that case continuing from the assessment dashboard would not have been possible, I think, as only the original assessor can do that.


> Canceling is readded :+1:

Thanks!